auth/routes/oauth_routes.go

```diff
@@ -6,10 +6,12 @@ const (
 	OAuthBegin    = "OAuthBegin"
 	OAuthCallback = "OAuthCallback"
 	LogOut        = "LogOut"
+	PostCustom    = "PostCustom"
 )
 
 var OAuthRoutes = rata.Routes{
 	{Path: "/auth/logout", Method: "GET", Name: LogOut},
 	{Path: "/auth/:provider/callback", Method: "GET", Name: OAuthCallback},
 	{Path: "/auth/:provider", Method: "GET", Name: OAuthBegin},
+	{Path: "/auth/:provider/", Method: "POST", Name: PostCustom},
```

Comment:
Yes, that is correct, with that we enable all auth providers to add additional routes that can be addressed as POST.
Currently only the `githubProvider` and the `gitlabProvider` are using that to provide an endpoint to authenticate with a token.
You can see that here: https://github.com/Chumper/atc/blob/4fb3dd3c9beba37e46a32967d7843b649b130077/auth/github/provider.go#L182

If no routes are added, a 404 will be thrown, so the default behaviour is the same.

The need to add routes is that by default only a webflow is allowed, that uses `GET auth/:provider` and the callback.